gads.py

```diff
@@ -20,38 +20,43 @@
 
 
 def get_gads_client(mcc_id: str) -> GoogleAdsClient:
-  main_creds = json.loads(get_value_session(session['session_id'], 'credentials'))
-  creds = {
-      "developer_token": os.environ["GOOGLE_ADS_DEVELOPER_TOKEN"],
-      "refresh_token": main_creds["refresh_token"],
-      "client_id": main_creds["client_id"],
-      "client_secret": main_creds["client_secret"],
-      "use_proto_plus": True,
-  }
+  try:
+    main_creds = json.loads(get_value_session(session['session_id'], 'credentials'))
+    creds = {
+        "developer_token": os.environ["GOOGLE_ADS_DEVELOPER_TOKEN"],
+        "refresh_token": main_creds["refresh_token"],
+        "client_id": main_creds["client_id"],
+        "client_secret": main_creds["client_secret"],
+        "use_proto_plus": True,
+    }
 
-  google_ads_client = GoogleAdsClient.load_from_dict(creds)
-  if mcc_id:
-    google_ads_client.login_customer_id = mcc_id
+    google_ads_client = GoogleAdsClient.load_from_dict(creds)
+    if mcc_id:
+        google_ads_client.login_customer_id = mcc_id
 
-  return google_ads_client
+    return google_ads_client
+  except:
+    return "Session Expired"
```

Comment:
so what code throws the exception here? is it the load_from_dict throwing a value error becuase the sessions is invalid?

Reply:
or maybe it's json.loads that throws an exception:
```
>>> x = '{"lol":1}'
>>> json.loads(x)
{'lol': 1}
>>> x = None
>>> json.loads(x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/__init__.py", line 341, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not NoneType
```

Reply:
JSON will throw it because the json code it is trying to load is no longer available, so tries to create a json credential file from nothing which is nontype and throws error.

Reply:
then the catch should only be on that line and check the exception or check if the session exists and return early